module/extension/FileWatcher/src/FileWatcher.cpp

```diff
@@ -290,7 +291,24 @@ namespace module::extension {
     }
 
     FileWatcher::~FileWatcher() {
+        for (const auto& path : paths) {
+            if (path.second.handle) {
+                uv_fs_event_stop(path.second.handle.get());
+                uv_close(reinterpret_cast<uv_handle_t*>(path.second.handle.get()), [](uv_handle_t*) {});
+            }
+        }
+
+        for (const auto& remove : remove_queue) {
+            if (remove) {
+                uv_fs_event_stop(remove.get());
+                uv_close(reinterpret_cast<uv_handle_t*>(remove.get()), [](uv_handle_t*) {});
+            }
+        }
+        uv_close(reinterpret_cast<uv_handle_t*>(remove_watch.get()), [](uv_handle_t*) {});
+        uv_close(reinterpret_cast<uv_handle_t*>(add_watch.get()), [](uv_handle_t*) {});
+        uv_close(reinterpret_cast<uv_handle_t*>(shutdown.get()), [](uv_handle_t*) {});
+        while (uv_run(loop.get(), UV_RUN_NOWAIT) != 0) {
+        }
```

Comment:
Does it work if you put this up the top at line 294? And if you do does it deal with the remove_queue?

Reply:
nope

Reply:
seems to hang

Reply:
Since someone made me remove 19 other files... i have 278 things to compile

Reply:
anywhere other than after closing the asyncs causes the test to hang (didn't test after the loop close)